pkg/scraper/client/resource/client.go

```diff
@@ -107,17 +107,20 @@ func (kc *kubeletClient) getMetrics(ctx context.Context, url, nodeName string) (
 	if response.StatusCode != http.StatusOK {
 		return nil, fmt.Errorf("request failed, status: %q", response.Status)
 	}
-	b := kc.buffers.Get().([]byte)
+	bp := kc.buffers.Get().(*[]byte)
+	b := *bp
```

Comment:
this shouldn't be necessary as far as I can tell and copying the content would allocate a new slice on the stack which kind of goes against the purpose of SA6002 to improve allocations.

Reply:
Copying the slice header to the stack does not _allocate_, as it's just 24 bytes copied to the stack. This cannot be avoided (whether we define a variable for this or not doesn't matter, it will be copied anyway when we create the bytes.Buffer below).

There are no heap memory allocations happening here.

Reply:
no sure where I was going with that one, sorry about that